src/main/java/com/commercetools/project/sync/category/CategorySyncer.java

```diff
@@ -80,22 +78,14 @@ public static CategorySyncer of(
 
     final CustomObjectService customObjectService = new CustomObjectServiceImpl(targetClient);
 
-    final CategoryReferenceTransformService referenceTransformService =
-        new CategoryReferenceTransformTransformServiceImpl(sourceClient);
-
-    return new CategorySyncer(
-        categorySync,
-        sourceClient,
-        targetClient,
-        customObjectService,
-        referenceTransformService,
-        clock);
+    return new CategorySyncer(categorySync, sourceClient, targetClient, customObjectService, clock);
   }
 
   @Override
   @Nonnull
   protected CompletionStage<List<CategoryDraft>> transform(@Nonnull final List<Category> page) {
-    return this.referencesService.transformCategoryReferences(page);
+    final ReferenceIdToKeyCache referenceIdToKeyCache = new CaffeineReferenceIdToKeyCacheImpl();
```

Comment:
Looks like the transform method will create a new cache for every call. Do you think this is the right purpose of caching? 🤔 

I think the `referenceIdToKeyCache` is in static scope but when I check the code my first impression above shows something wrong about its design. How we could improve that ?

Reply:
Shall i add it into the [Syncer](https://github.com/commercetools/commercetools-project-sync/blob/bcc420837079cd674a521dc326ac010e2e997994/src/main/java/com/commercetools/project/sync/Syncer.java#L63) class as a property and set the same cache for all the resources? 
What do you think?
I tried doing this and commit here but there were so many changes, So will discuss first and then I will go with the approach.

Reply:
What do you mean by so many changes? Yes, you could add it as a static protected instance of ReferenceIdToKeyCache in Syncer.

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/268/commits/2cea71e21f130b4e01ada411ec43d4d8c35792f5